peps/pep-0733.rst

```diff
@@ -0,0 +1,629 @@
+[`Issue 6 <https://github.com/capi-workgroup/problems/issues/6>`__].
+
+The fact that ``PyArg_ParseTupleAndKeywords`` takes a non-const
+``char*`` array as argument makes it more difficult to use
+[`Issue 28 <https://github.com/capi-workgroup/problems/issues/28>`__].
+
+``Python.h`` does not expose the whole API. Some headers (like ``marshal.h``)
+are not included from ``Python.h``.
+[`Issue 43 <https://github.com/capi-workgroup/problems/issues/43>`__].
+
+**Naming**
+
+``PyLong`` and ``PyUnicode`` use names which don't match the Python
+types they represent (``int``/``str``). This can be fixed in a new API
+[`Issue 14 <https://github.com/capi-workgroup/problems/issues/14>`__].
+
+There are identifiers in the API which are lacking a ``Py``/``_Py``
+prefix
+[`Issue 46 <https://github.com/capi-workgroup/problems/issues/46>`__].
+
+Missing Functionality
+---------------------
+
+This section consists of a list of feature requests, i.e., functionality
+that was identified as missing in the current C API.
+
+Debug Mode
+~~~~~~~~~~
+
+A debug mode that can be activated without recompilation and which
+activates various checks that can help detect various types of errors
+[`Issue 36 <https://github.com/capi-workgroup/problems/issues/36>`__].
+
+Introspection
+~~~~~~~~~~~~~
+
+There aren't currently reliable introspection capabilities for objects
+defined in C in the same way as there are for Python objects
+[`Issue 32 <https://github.com/capi-workgroup/problems/issues/32>`__].
+
+Efficient type checking for heap types, similar to what ``Py*_Check``
+can do for a static type
+[`Issue 17 <https://github.com/capi-workgroup/problems/issues/17>`__].
+
+Improved Interaction with Other Languages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Interfacing with other GC based languages, and integrating their
+GC with Python's GC
+[`Issue 19 <https://github.com/capi-workgroup/problems/issues/19>`__].
+
+Inject foreign stack frames to the traceback
+[`Issue 18 <https://github.com/capi-workgroup/problems/issues/18>`__].
+
+Concrete strings that can be used in other languages
+[`Issue 16 <https://github.com/capi-workgroup/problems/issues/16>`__].
+
+References
+==========
+
+1. `Python/C API Reference Manual <https://docs.python.org/3/c-api/index.html>`__
+2. `2023 Language Summit Blog Post: Three Talks on the C API <https://pyfound.blogspot.com/2023/05/the-python-language-summit-2023-three.html>`__
+3. `capi-workgroup on GitHub <https://github.com/capi-workgroup>`__
+4. 🔒 `Irit's Discord post with Core Sprint 2023 slides about C API workgroup <https://discord.com/channels/854719841091715092/897862918115315813/1160952205801689249>`__
+5. 🔒 `Petr's Discord post with Core Sprint 2023 slides <https://discord.com/channels/854719841091715092/897862918115315813/1160939833473126411>`__
+6. 🔒 `Steve's Discord post with Core Sprint 2023 slides for Things to Learn from HPy <https://discord.com/channels/854719841091715092/897862918115315813/1160942893696024578>`__
```

Comment:
These are not public links - will only be accessible by those with permission. Is that an okay reference to include in a public doc?

Reply:
Irit's and Petr's slides are public and the links have been updated:

https://pep-previews--3491.org.readthedocs.build/pep-0733/#references

And the Hpy slides will be public soon.